lib/components/Charts/BaseBarChart/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { ChartContainer } from "@/ui/chart"
+import { ForwardedRef } from "react"
+import { ChartConfig, ChartPropsBase, InferChartKeys } from "../utils/types"
+
+import React from "react"
+import { Bar, ComposedChart, LabelList } from "recharts"
+import { prepareData } from "../utils/bar"
+import { fixedForwardRef } from "../utils/forwardRef"
+
+type BaseBarChartConfigProps = {
+  layout?: "vertical" | "horizontal"
+  xAxisComponent?: React.ReactElement
+  yAxisComponent?: React.ReactElement
+  gridComponent?: React.ReactElement
+}
+
+export type BaseBarChartProps<
+  DataConfig extends ChartConfig = ChartConfig,
+  Keys extends string = InferChartKeys<DataConfig>,
+> = ChartPropsBase<DataConfig, Keys> & { label: boolean }
+
+export const _BaseBarChart = <
+  DataConfig extends ChartConfig,
+  Keys extends string = string,
+>(
+  config: BaseBarChartProps<DataConfig, Keys> & BaseBarChartConfigProps,
+  ref: ForwardedRef<HTMLDivElement>
+) => {
+  const {
+    dataConfig,
+    data,
+    label,
+    xAxisComponent,
+    yAxisComponent,
+    layout,
+    gridComponent,
+  } = config
+  const bars = Object.keys(dataConfig) as Array<keyof typeof dataConfig>
+
+  return (
+    <ChartContainer config={dataConfig} ref={ref}>
+      <ComposedChart
+        layout={layout}
+        accessibilityLayer
+        data={prepareData(data)}
+        margin={{ left: 12, right: 12 }}
+      >
+        {gridComponent}
+        {xAxisComponent}
+        {yAxisComponent}
+
+        {bars.map((l, lt) => {
+          return (
+            <>
+              <Bar
+                layout={layout}
+                key={`line-${lt}-lt`}
+                dataKey={l}
+                fill={dataConfig[l].color}
+                radius={4}
+              >
+                {label && (
+                  <LabelList
+                    key={`label-{${lt}}`}
+                    position={layout == "vertical" ? "right" : "top"}
+                    offset={10}
+                    className="fill-foreground"
+                    fontSize={12}
+                  />
+                )}
+              </Bar>
+            </>
+          )
+        })}
+      </ComposedChart>
+    </ChartContainer>
+  )
+}
+
+_BaseBarChart.defaultProps = {
+  layout: "horizontal",
+}
+
+export const BarChart = fixedForwardRef(_BaseBarChart)
```

Comment:
I feel we should somehow clarify that this is private. Maybe if we create a `BarChart` folder that contains both and this within with a `_` prepended? Also it might not need its own folder, as it doesn't have stories. That's the convention I'm generally pushing for for private classes, but maybe it's not the best.

Reply:
@josepjaumeI was concerned about repeating the same code in two places. Having duplicated code could become problematic if updates are needed in the future, as changes would have to be made in multiple locations. Therefore, I initially considered a middle ground between duplicating the code and creating a single component to manage the layout.

However, since there are currently only two components and adding complexity to the logic isn't ideal, I've decided to keep the bar components without any abstraction.